### PR TITLE
[SPARK-5248] [SQL] move sql.types.decimal.Decimal to sql.types.Decimal

### PR DESCRIPTION
**Pull request by adrian-wang**
@rxin follow up of #3732

### CODE REVIEW

**Comment by adrian-wang** [comment]
retest this please.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlSerializer.scala [hunk 1]
```diff
@@ -19,6 +19,8 @@ package org.apache.spark.sql.execution
 
 import java.nio.ByteBuffer
 
+import org.apache.spark.sql.types.Decimal
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlSerializer.scala]: import out of order here...

File: sql/catalyst/src/test/scala/org/apache/spark/sql/types/decimal/DecimalSuite.scala [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 package org.apache.spark.sql.types.decimal
 
+import org.apache.spark.sql.types.Decimal
```
Comment (rxin) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/types/decimal/DecimalSuite.scala]: import out of order here... it's ok i will fix it myself.

**Comment by rxin** [comment]
Thanks. Going to merge this.

**Pull request closed without merging**